tests_e2e/tests/ext_sequencing/ext_seq_test_cases.py

```diff
@@ -0,0 +1,300 @@
+def add_one_dependent_ext_without_settings():
+    # Dependent extensions without settings should be enabled with dependencies
+    return [
+        {
+            "name": "AzureMonitorLinuxAgent",
+            "properties": {
+                "provisionAfterExtensions": ["CustomScript"],
+                "publisher": "Microsoft.Azure.Monitor",
+                "type": "AzureMonitorLinuxAgent",
+                "typeHandlerVersion": "1.5",
+                "autoUpgradeMinorVersion": True
+            }
+        },
+        {
+            "name": "CustomScript",
+            "properties": {
+                "publisher": "Microsoft.Azure.Extensions",
+                "type": "CustomScript",
+                "typeHandlerVersion": "2.1",
+                "autoUpgradeMinorVersion": True,
+                "settings": {}
+            }
+        }
+    ]
+
+
+def add_two_extensions_with_dependencies():
+    return [
+        {
+            "name": "AzureMonitorLinuxAgent",
+            "properties": {
+                "provisionAfterExtensions": [],
+                "publisher": "Microsoft.Azure.Monitor",
+                "type": "AzureMonitorLinuxAgent",
+                "typeHandlerVersion": "1.5",
+                "autoUpgradeMinorVersion": True,
+                "settings": {}
+            }
+        },
+        {
+            "name": "RunCommandLinux",
+            "properties": {
+                "provisionAfterExtensions": ["AzureMonitorLinuxAgent"],
+                "publisher": "Microsoft.CPlat.Core",
+                "type": "RunCommandLinux",
+                "typeHandlerVersion": "1.0",
+                "autoUpgradeMinorVersion": True,
+                "settings": {}
+            }
+        },
+        {
+            "name": "CustomScript",
+            "properties": {
+                "provisionAfterExtensions": ["RunCommandLinux", "AzureMonitorLinuxAgent"],
+                "publisher": "Microsoft.Azure.Extensions",
+                "type": "CustomScript",
+                "typeHandlerVersion": "2.1",
+                "autoUpgradeMinorVersion": True,
+                "settings": {}
+            }
+        }
+    ]
+
+
+def remove_one_dependent_extension():
```

Comment:
this case seems to have a dependency on another test case that adds the extension we are removing. let's add that as a comment in the ExtSequencing.test_cases

Reply:
similar comments in some of the cases below